src/cache_cf.cc

```diff
@@ -3540,7 +3540,11 @@ parse_port_option(AnyP::PortCfgPointer &s, char *token)
             return;
         }
         s->flags.natIntercept = true;
-        Ip::Interceptor.StartInterception();
+        if (!Ip::Interceptor.StartInterception()) {
+            debugs(3, DBG_CRITICAL, "FATAL: " << cfg_directive << " requires intercepted proxying enabled by ./configure");
+            self_destruct();
+            return;
+        }
```

Comment:
I am not against making StartInterception() throw an exception instead. However, it would be effectively the same functionality change resulting in killing Squid during startup.

Reply:
@yadij, the manner of Squid startup death (exception vs self_destruct()) aside, I assume that you are OK with this functionality change in this PR. As you know, I am all for refusing (to run with) half-broken configurations, so I am happy to use this opportunity to improve things. However, please correct me if my assumption is wrong (and Eduard should remove this functionality change from this PR instead of polishing it).

@eduard-bagdasaryan, if throwing is not making things _worse_ than calling self_destruct(), then we should throw (for many reasons). FWIW, when I see "added" code in with self_destruct() calls in the diff, I assume that this is a mistake or moved code. New code should throw. (Moved code may be converted to throwing as well, but we need to balance many factors when moving code so the decision is more nuanced when moving...)